releasenotes/notes/add-parameters-to-template-substitution-a1379cdbfcc10b5c.yaml

```diff
@@ -0,0 +1,88 @@
+    implemented for the RXGate().
+issues:
+  - |
+    Currently the only gate that can be inverted (as done in the template 
+    optimization algorithm) with a Parameter is the RXGate().
+other:
+  - |
+    An illustrative example of this PR is modeled after the addition of 
+    calibrations for template optimization #5752:
+
+    .. code-block:: 
+
+      from qiskit import QuantumCircuit, transpile, schedule, IBMQ
+      from qiskit.circuit import Parameter
+      
+      from qiskit.transpiler import PassManager
+      from qiskit.transpiler.passes import TemplateOptimization
+      
+      # New contributions to the template optimization
+      from qiskit.transpiler.passes.scheduling import RZXCalibrationBuilder, rzx_templates
+      
+      IBMQ.load_account()
+      provider = IBMQ.get_provider(...)
+      backend = provider.get_backend('ibmq_casablanca')
+      
+      phi = Parameter('φ')
+      
+      qc = QuantumCircuit(2)
+      qc.cx(0,1)
+      qc.p(2*phi, 1)
+      qc.cx(0,1)
+      print('Original circuit:')
+      print(qc)
+      
+      pass_ = TemplateOptimization(**rzx_templates(['zz2']))
+      qc_cz = PassManager(pass_).run(qc)
+      print('ZX based circuit:')
+      print(qc_cz)
+      
+      # Add the calibratiocns
+      pass_ = RZXCalibrationBuilder(backend)
+      cal_qc = PassManager(pass_).run(qc_cz.bind_parameters({phi: 0.12}))
+      
+      # Transpile to the backend basis gates
+      cal_qct = transpile(cal_qc, backend)
+      qct = transpile(qc.bind_parameters({phi: 0.12}), backend)
+      
+      # Compare the schedule durations
+      print('Duration of schedule with the calibration:')
+      print(schedule(cal_qct, backend).duration)
+      print('Duration of standard with two CNOT gates:')
+      print(schedule(qct, backend).duration)
+          
+    outputs
+
+    .. parsed-literal::
+
+      Original circuit:
+      			 
+      q_0: ──■──────────────■──
+           ┌─┴─┐┌────────┐┌─┴─┐
+      q_1: ┤ X ├┤ P(2*φ) ├┤ X ├
+           └───┘└────────┘└───┘
+      ZX based circuit:
+      					 ┌─────────────┐            »
+      q_0: ────────────────────────────────────┤0            ├────────────»
+           ┌──────────┐┌──────────┐┌──────────┐│  Rzx(2.0*φ) │┌──────────┐»
+      q_1: ┤ Rz(-π/2) ├┤ Rx(-π/2) ├┤ Rz(-π/2) ├┤1            ├┤ Rx(-2*φ) ├»
+           └──────────┘└──────────┘└──────────┘└─────────────┘└──────────┘»
+      «                                                     
+      «q_0: ────────────────────────────────────────────────
+      «     ┌──────────┐┌──────────┐┌──────────┐┌──────────┐
+      «q_1: ┤ Rz(-π/2) ├┤ Rx(-π/2) ├┤ Rz(-π/2) ├┤ P(2.0*φ) ├
+      «     └──────────┘└──────────┘└──────────┘└──────────┘
+      Duration of schedule with the calibration:
+      1600
+      Duration of standard with two CNOT gates:
+      6848
+      
+    Note that the Parameters/ParameterExpressions must be bound before the 
+    RZXCalibrationBuilder can define calibrations for the RZXGates.
```

Comment:
The release notes for different sections aren't grouped, so referring to "this PR" will lose the context.  Examples for the PR can go with the feature note.

We don't need to explain where the example came from, just the example is enough.  It'd be good to remove the dependence on `IBMQ` and `load_account` so we're not executing network code on users' hardware if they choose to run this; it'd be better to use a mock backend (e.g. `from qiskit.test.mock import FakeCasablanca`).

Reply:
Fixed the release notes to just give the example, include only a mock backend, eliminate the issues that have since been resolved, and update some transpiler imports.